src/encoderust.rs

```diff
@@ -30,7 +30,11 @@ impl<W: Write> Encoder for RustEncoder<W> {
                 let w = writer.take().expect("writer");
                 let mut enc = gif::Encoder::new(w, screen_width, screen_height, &[])?;
                 if !settings.once {
-                    enc.write_extension(gif::ExtensionData::Repetitions(gif::Repeat::Infinite))?;
+                    if settings.loop_count > 0 {
+                        enc.write_extension(gif::ExtensionData::Repetitions(gif::Repeat::Finite(settings.loop_count)))?;
+                    } else {
+                        enc.write_extension(gif::ExtensionData::Repetitions(gif::Repeat::Infinite))?;
+                    }
```

Comment:
```suggestion
                    let repetitions = settings.loop_count > 0 ? gif::Repeat::Finite(settings.loop_count) : gif::Repeat::Infinite;
                    enc.write_extension(gif::ExtensionData::Repetitions(repetitions))?;
```

Not sure if this is valid Rust code, but reads a bit clearer to me.